src/components/pages/home_screen/task-detail/TaskDetail.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useEffect } from "react";
+import {
+  TextField,
+  Button,
+  Box,
+  MenuItem,
+  Typography,
+  IconButton,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { Task, Category } from "../types";
+import { addDoc, collection, doc, updateDoc } from "firebase/firestore";
+import { db, auth } from "../../../firebase";
+import "./_taskDetail.scss";
+
+interface TaskDetailProps {
+  selectedTask: Task | null;
+  categories: Category[];
+  updateTask: (task: Task) => void;
+  addTask: (task: Task) => void;
+  setSelectedTask: (task: Task | null) => void;
+  deleteTask: (id: string) => void;
+  closeTaskDetail: () => void;
+}
+
+const TaskDetail = ({
+  selectedTask,
+  categories,
+  updateTask,
+  addTask,
+  setSelectedTask,
+  closeTaskDetail,
+}: TaskDetailProps) => {
+  const [task, setTask] = useState<Task | null>(selectedTask);
+
+  useEffect(() => {
+    setTask(
+      selectedTask || {
+        id: "",
+        title: "",
+        description: "",
+        dueDate: "",
+        list: "",
+        completed: false,
+      }
+    );
+  }, [selectedTask]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setTask((prev) => (prev ? { ...prev, [name]: value } : null));
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (task) {
+      if (task.id) {
```

Comment:
Check trực tiếp luôn e 
```suggestion
    if (task && task.id) {
       ....
    } else {
       ....
    }
```